### DISCUSSION TITLE
A stricter ipv4 format

### DISCUSSION THREAD

**Discussion by Chardrazle** [json-schema-org/community#271] [Specification]
The current `ipv4` format type allows zero-prefixes in the octets.
When these are consumed by whatever code requires the field they can often go through API functions that treat the zero-prefix octet as an _octal_ value.  This can lead to obvious misconfiguration.

Although it's relatively trivial to provide a custom pattern, it would be nice if a stricter ipv4 formatter was available in the specification.

**Comment by handrews** [comment]
`format` just defers to other specifications, and dotted-quad is standard for ip4v.  Code should understand to parse dotted-quad accordingly and not as octal digits.

**Comment by Chardrazle** [comment]
Sure, format/[ipv4](https://www.rfc-editor.org/rfc/rfc2673#section-3.2) is following a particular standard.  However, because of different API implementations the code is often forced to strip out any leading prefixes from each IPv4 part.
For example, a C implementation may be throwing the input at [inet_addr](https://pubs.opengroup.org/onlinepubs/009695399/functions/inet_addr.html), which would treat leading zeros as octal.
A Python implementation may throw the input at [ipaddress](https://docs.python.org/3/library/ipaddress.html), which flatly rejects leading zeros (due to the octal issue).
To avoid any issue, it's often easier to simply reject the leading zeros at the data-input layer.

**Reply by handrews** [reply]
That's really a regular expression thing.  We're not going to specify a `format` with non-standard behavior, it's just too likely to cause problems.

Of course, you can also make your own extension vocabulary with keywords that do what you want (technically you can make your own `format` values, too, but the support for that is pretty inconsistent)
